src/display/annotation_layer.js

```diff
@@ -67,7 +67,16 @@ AnnotationElementFactory.prototype =
         return new TextAnnotationElement(parameters);
 
       case AnnotationType.WIDGET:
-        return new WidgetAnnotationElement(parameters);
+        switch (parameters.data.fieldType) {
+          case 'Tx':
+            return new TextWidgetAnnotationElement(parameters);
+          default:
+            warn('Unimplemented Widget annotation type: ' +
+                  parameters.data.fieldType);
+            // Fallback to default one, which does not render anything
```

Comment:
Also, this block seems unnecessarily complicated. Wouldn't e.g. this work just as well:
```diff
-  return new WidgetAnnotationElement(parameters);
+  if (parameters.data.fieldType === 'Tx') {
+    return new TextWidgetAnnotationElement(parameters);
+  }
+  warn('Unimplemented Widget annotation type: ' +
+       parameters.data.fieldType);
+  // Fallback to default one, which does not render anything.
+  return new WidgetAnnotationElement(parameters);

```